### PR TITLE
added permission search

### PR DESCRIPTION
**Pull request by edewit**
added the search bar of the authorization search permissions

### CODE REVIEW

File: src/clients/authorization/Permissions.tsx [hunk 1]
```diff
@@ -166,10 +170,12 @@ export const AuthorizationPermissions = ({ clientId }: PermissionsProps) => {
     return <KeycloakSpinner />;
   }
 
+  const noData = permissions.length === 0;
+  const searching = !isEqual(search, {});
```
Comment (jonkoops) [inline src/clients/authorization/Permissions.tsx:174]: I think you could accomplish the same without using Lodash by doing the following:
```suggestion
  const searching = Object.keys(search).length !== 0;
```

File: src/clients/authorization/SearchDropdown.tsx [hunk 2]
```diff
@@ -69,24 +108,46 @@ export const SearchDropdown = ({ types }: SearchDropdownProps) => {
...
-                {types?.map((type) => (
-                  <SelectOption
-                    selected={type.type === value.type}
-                    key={type.type}
-                    value={type}
-                  >
-                    {type.name}
-                  </SelectOption>
-                ))}
+                {[
```
Comment (jonkoops) [inline src/clients/authorization/SearchDropdown.tsx:116]: This is perhaps a place you could use a fragment? Or does that bork the control from PatternFly?

**Review by jonkoops — APPROVED** [review]

File: src/clients/authorization/SearchDropdown.tsx (hunk 2 above)
Comment (edewit) [reply src/clients/authorization/SearchDropdown.tsx:116]: No, but we have the rule not to have a single child in a fragment right?

Comment (jonkoops) [reply src/clients/authorization/SearchDropdown.tsx:116]: In this case it would not be a single child as there are possibly multiple options mapped below.

Comment (jonkoops) [reply src/clients/authorization/SearchDropdown.tsx:116]: Why is this an array of JSX elements in this case? What prevents us from doing:

```jsx
<Select>
  <SelectOption>Default value</SelectOption>
  {otherOptions.map((name) => (<SelectOption>{name}</SelectOption>))
</Select>
```

Comment (edewit) [reply src/clients/authorization/SearchDropdown.tsx:116]: Types prevent us from doing that.. seems that patternfly only wants one child

Comment (edewit) [reply src/clients/authorization/SearchDropdown.tsx:116]: how about:
```typescript
  const typeOptions = (value: string) => [
    <SelectOption key="empty" value="">
      {t("allTypes")}
    </SelectOption>,
    ...(types || []).map((type) => (
      <SelectOption
        selected={type.type === value}
        key={type.type}
        value={type.type}
      >
        {type.name}
      </SelectOption>
    )),
  ];

// ...

            <Select
                toggleId="type"
                onToggle={toggleType}
                onSelect={(event, value) => {
                  event.stopPropagation();
                  onChange(value);
                  toggleType();
                }}
                selections={value || t("allTypes")}
                variant={SelectVariant.single}
                aria-label={t("common:type")}
                isOpen={typeOpen}
              >
                {typeOptions(value)}
            </Select>
```

**Pull request merged**